cmd/relayproxy/controller/collect_eval_data.go

```diff
@@ -47,6 +47,7 @@ func (h *collectEvalData) Handler(c echo.Context) error {
 	}
 
 	for _, event := range reqBody.Events {
+		event.Source = "PROVIDER_CACHE"
```

Comment:
I think we should not override the `Source` if it is already provided in the `event`.
What do you think if we have something like this instead:
```suggestion
		if event.Source == "" {		
		    event.Source = "PROVIDER_CACHE"
		}
```

Reply:
Do we also want to check that when the source is SERVER? Trying to think of an use case in which that could be useful 🤔

Reply:
Yes good point.
Be awesome if you could do it.

i can help with that if you want.

Reply:
this one seems a little bit trickier

well, if you have time, help is always welcome :slightly_smiling_face:, but don't worry if you don't, I can give it a try

Reply:
I was thinking about it again this morning, and I think this use case is not really possible (overriding when in `SERVER` mode).